src/Amazon.Common.DotNetCli.Tools/Utilities.cs

```diff
@@ -205,72 +206,129 @@ public static string DeterminePublishLocation(string workingDirectory, string pr
 
-            var xdoc = XDocument.Load(projectFile);
-
-            var element = xdoc.XPathSelectElement("//PropertyGroup/TargetFramework");
-            return element?.Value;
-        }
-
-        /// <summary>
-        /// Retrieve the `OutputType` property of a given project
+        // <summary>
+        /// Looks up specified properties from a project file.
         /// </summary>
-        /// <param name="projectLocation">Path of the project</param>
-        /// <returns>The value of the `OutputType` property</returns>
-        public static string LookupOutputTypeFromProjectFile(string projectLocation)
+        /// <param name="projectLocation">The location of the project file.</param>
+        /// <param name="propertyNames">The names of the properties to look up.</param>
+        /// <returns>A dictionary of property names and their values.</returns>        
+        public static Dictionary<string, string> LookupProjectProperties(string projectLocation, params string[] propertyNames)
         {
-            var process = new Process();
-            var output = string.Empty;
-            var msbuildProcessFailed = false;
-            try
+            var projectFile = FindProjectFileInDirectory(projectLocation);
+            var properties = new Dictionary<string, string>();
+            var arguments = new List<string>
             {
-                process.StartInfo = new ProcessStartInfo()
+                "msbuild",
+                projectFile,
+                "-nologo",
+                $"--getProperty:{string.Join(',', propertyNames)}"
+            };
+
+            var process = new Process
+            {
+                StartInfo = new ProcessStartInfo
                 {
                     FileName = "dotnet",
-                    Arguments = $"msbuild {projectLocation} -getProperty:OutputType",
+                    Arguments = string.Join(" ", arguments),
                     RedirectStandardOutput = true,
+                    RedirectStandardError = true,
                     UseShellExecute = false,
-                    WindowStyle = ProcessWindowStyle.Hidden
-                };
-
+                    CreateNoWindow = true
+                }
+            };
+            try
+            {
                 process.Start();
-                output = process.StandardOutput.ReadToEnd();
-                var hasExited = process.WaitForExit(5000);
+                string output = process.StandardOutput.ReadToEnd();
+                string error = process.StandardError.ReadToEnd();
+                process.WaitForExit(5000);
 
-                // If it hasn't completed in the specified timeout, stop the process and give up
-                if (!hasExited) 
+                if (process.ExitCode == 0)
                 {
-                    process.Kill();
-                    msbuildProcessFailed = true;
-                }
+                    using JsonDocument doc = JsonDocument.Parse(output);
```

Comment:
I might be missing something but if propertyNames is a single value isn't the output of the command just the value and not the JSON document?

Reply:
```
PS C:\dev\repos\aws-extensions-for-dotnet-cli> dotnet msbuild .\testapps\TestFunction\TestFunction.csproj  --getProperty:TargetFramework,TargetFrameworks
{
  "Properties": {
    "TargetFramework": "net6.0",
    "TargetFrameworks": ""
  }
}
```

is what the commmand returns

Reply:
We need to check if the count of propertyNames is 1 or greater since the result varies. If you run:
```
dotnet msbuild --nologo --getProperty:TargetFramework
```
The result will be 
```
net8.0
```
If you run
```
dotnet msbuild --nologo --getProperty:TargetFramework,TargetFrameworks
```
The result will be:
```
{
  "Properties": {
    "TargetFramework": "net8.0",
    "TargetFrameworks": ""
  }
}
```
You need to handle both cases

Reply:
ah yes you are right. i will update that!

Reply:
updated to handle case for when only one property is provided. Thanks for catching this!